File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/lombok/allArgsConstructor.java
// !ADD_LOMBOK_ANNOTATIONS
// KTIJ-19423: @AllArgsConstructor takes every field, not only the required ones
package test;

import lombok.AllArgsConstructor;
import lombok.NonNull;

@AllArgsConstructor
public class Everything {
    private final String required;

    private String mutable;

    @NonNull
    private String nonNull;

    private final String initialized = "skipped";

    private static String shared;
}

@AllArgsConstructor
class NoInstanceFields {
    private static final String CONSTANT = "c";
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/lombok/allArgsConstructor.kt
// !ADD_LOMBOK_ANNOTATIONS
// KTIJ-19423: @AllArgsConstructor takes every field, not only the required ones
package test

class Everything(private val required: String, private val mutable: String?, private val nonNull: String) {
    private val initialized = "skipped"

    companion object {
        private val shared: String? = null
    }
}

internal object NoInstanceFields {
    private const val CONSTANT = "c"
}
